ionoscloud/resource_server_vcpu.go

```diff
@@ -0,0 +1,365 @@
+package ionoscloud
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+func resourceVCPUServer() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVCPUServerCreate,
+		ReadContext:   resourceServerRead,
+		UpdateContext: resourceServerUpdate,
+		DeleteContext: resourceServerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceServerImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			"cores": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"ram": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"availability_zone": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"AUTO", "ZONE_1", "ZONE_2"}, true)),
+			},
+			"boot_volume": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"boot_cdrom": {
```

Comment:
add a description and the is UUID validation please

Reply:
@cristiGuranIonos done in 52c9a9c839f1d083097f21fbd6777ca8c3fbfb14